### PR TITLE
Pack for SignalR push notification system

### PR DESCRIPTION
**Pull request by tonybaloney**
Allows push notifications to be sent to a SignalR message hub and those messages to be received by StackStorm and converted into triggers.
Messages can also be sent back to the hub via an action.

### CODE REVIEW

File: packs/signalr/actions/send_message.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class SignalRSendMessageAction(Action):
+    def __init__(self, config=None):
+        super(SignalRSendMessageAction, self).__init__(config=config)
+        self.url = config['hub_url']
+        self.session = None
+
+    def run(self, hub, message):
+        connection = Connection(self.url, self.session)
```
Comment (lakshmi-kannan) [inline packs/signalr/actions/send_message.py]: session is None? Is this right?

File: packs/signalr/sensors/signalr_hub_sensor.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        self.connection = None
+        self.session = None
+
+    def setup(self):
+        self.connection = Connection(self.url, self.session)
+        # start a connection
+        self.connection.start()
+        # add a handler to process notifications to the connection
+        self.connection.handlers += \
+            lambda data: self._logger.log(
```
Comment (lakshmi-kannan) [inline packs/signalr/sensors/signalr_hub_sensor.py]: I am pretty sure there is no `log` method. You want `debug` or `info` I'd think.

File: packs/signalr/sensors/signalr_hub_sensor.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        self._logger = self._sensor_service.get_logger(__name__)
+        self.url = config['hub_url']
+        self.hub_name = config['hub_name']
+        self._trigger_ref = 'signalr.message_received'
+        self._hub = None
+        self.connection = None
+        self.session = None
+
+    def setup(self):
+        self.connection = Connection(self.url, self.session)
```
Comment (lakshmi-kannan) [inline packs/signalr/sensors/signalr_hub_sensor.py]: Same q about session=None

File: packs/signalr/actions/send_message.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        connection.start()
+        # get hub
+        _hub = self.connection.hub(hub)
```
Comment (lakshmi-kannan) [inline packs/signalr/actions/send_message.py]: You could just reassign hub. _ is a little sketch.

File: packs/signalr/sensors/signalr_hub_sensor.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        # start a connection
+        self.connection.start()
+        # add a handler to process notifications to the connection
+        self.connection.handlers += \
+            lambda data: self._logger.log(
+                'Connection: new notification - %s' % data)
+        # get hub
+        self._hub = self.connection.hub(self.hub_name)
+
+    def message_recieved(self, message):
```
Comment (lakshmi-kannan) [inline packs/signalr/sensors/signalr_hub_sensor.py]: Typo in method name. Linter caught it.

File: packs/signalr/sensors/signalr_hub_sensor.py (hunk 2 above)
Comment (Kami) [inline packs/signalr/sensors/signalr_hub_sensor.py]: There actually is - it's the most generic one which requires you to specify level (other methods are just wrappers around this one with the level already provided).

**Pull request merged**